flutter/ios/Classes/SwiftSentryFlutterPlugin.swift

```diff
@@ -3,27 +3,257 @@ import Sentry
+            }
+
+            if let maxBreadcrumbs = arguments["maxBreadcrumbs"] as? UInt{
+                options.maxBreadcrumbs = maxBreadcrumbs
+            }
+
+            print("------ options.integrations =======> \(String(describing: options.integrations))")
+            //print("------ packages =======> \(String(describing: arguments["packages"] as? [String]))")
+            //print("------ integrations =======> \(String(describing: arguments["integrations"] as? [String]))")
+
+            self.sentryOptions = options
+            /*
+             TODO : beforeSend alternative =>
+              - eventOrigin
+             */
+
+            /* for now, in sentry-cocoa, beforeSend is not called before captureEnvelope
+            options.beforeSend = { event in
+                self.setEventOriginTag(event: event)
+
+                if let sdk = event.sdk, self.isValidSdk(sdk: sdk){
+                    if let packages = arguments["packages"] as? [String]{
+                        if  var sdkPackages = sdk["packages"] as? [String]{
+                            event.sdk!["packages"] = sdkPackages.append(contentsOf: packages)
+                        } else {
+                            event.sdk = ["packages":packages]
+                        }
+                    }
+
+                    if let integrations = arguments["integrations"] as? [String]{
+                        if  var sdkIntegrations = sdk["integrations"] as? [String]{
+                            event.sdk!["integrations"] = sdkIntegrations.append(contentsOf: integrations)
+                        } else {
+                            event.sdk = ["integrations":integrations]
+                        }
+                    }
+                }
+                return event
+            }*/
+        }
+
+        result("iOS initNativeSdk" )
+    }
+
+    private func setEventOriginTag(event: Event){
+        guard let sdk = event.sdk else { return  }
+        if self.isValidSdk(sdk:sdk){
+
+            switch sdk["name"] as! String{
+            case "sentry.dart.flutter":
+                setEventEnvironmentTag(event:event,origin:"flutter",environment:"dart")
+            case "sentry.cocoa":
+                setEventEnvironmentTag(event:event,origin: "flutter",environment: "dart")
+            case "sentry.native":
+                setEventEnvironmentTag(event:event,origin: "flutter",environment: "dart")
+            default:
+                return
+            }
+        }
+    }
+
+    private func setEventEnvironmentTag(event: Event, origin: String = "ios", environment: String) {
+        event.tags?["event.origin"] = origin
+        event.tags?["event.environment"] = environment
+    }
+
+    private func isValidSdk( sdk: [String: Any]) -> Bool{
+        return (sdk["name"] != nil && !(sdk["name"] as! String).isEmpty)
+    }
+
+    func convertStringToDictionary(text: String) -> [String:AnyObject]? {
+        if let data = text.data(using: .utf8) {
```

Comment:
an `else` and a log would be nice, which means the envelope was not a `jsonObject` or not a `Map`

Reply:
✅